### PR TITLE
DA Kzgrs Backend Certificate implementation

### PR DESCRIPTION
**Pull request by bacv**
This PR implements certificate build and verification methods defined in [nomos-spec/dispersal](https://github.com/logos-co/nomos-specs/blob/master/da/dispersal.py).
VID and Metadata traits are also implemented for future use in the verifier, mempool and indexer services.

### CODE REVIEW

File: nomos-da/kzgrs-backend/src/dispersal.rs
Comment (danielSanchezQ) [inline nomos-da/kzgrs-backend/src/dispersal.rs:216]: Cant this be directly derived? 🤔

File: nomos-da/kzgrs/src/common.rs
Comment (danielSanchezQ) [inline nomos-da/kzgrs/src/common.rs:59]: Nice stuff!

File: nomos-da/kzgrs-backend/src/dispersal.rs
Comment (bacv) [reply nomos-da/kzgrs-backend/src/dispersal.rs]: Gave it another go and it works, I guess last time I've just added one of them to derive statement. Fixed

**Review by danielSanchezQ — APPROVED** [review]
Let's wait for @megonen approval before merging.

**Comment by bacv** [comment]
After the rebase on `da-protocol-v1` ([verifier integration tests recently merged](https://github.com/logos-co/nomos-node/pull/650)) the attestation was changed to hold Signature as `Vec<u8>` - this required to change the Certificate building code (minimally): https://github.com/logos-co/nomos-node/pull/651/files#diff-69966ce62bc67f8e54f7e1edfa1df7748dd18c14dae98da2d2276cf4e0055d12R67-R79

I've also added the same threshold used to check the number of received attestations to valid signatures, please review.

**Review by megonen — APPROVED** [review]

**Pull request merged**
